src/index.js

```diff
@@ -1 +1,32 @@
-export default () => true
+import defaultsDeep from 'lodash.defaultsdeep';
+import pickBy from 'lodash.pickby';
+
+class Stipulate {
+
+  constructor(options) {
+    this.baseOptions = options;
```

Comment:
I agree on the `baseUrl` option. It's something I definitely see us using a lot. If it's not present, I'd extend the class and add it in `beforeRequest` everytime.